New Catalan Cheat Sheet

rubikkk opened the pull request:
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [X] New Instant Answer
    - [X] Cheat Sheets: **`New {Cheat Sheet Name} Cheat Sheet`**
    - [ ] Other: **`New {IA Name} Instant Answer`**
- [ ] Improvement
    - [ ] Bug fix: **`{IA Name}: Fix {FIX typos}`**
    - [ ] Enhancement: **`{IA Name}: {Adding Food and drink}`**
- [ ] Non–Instant Answer
    - [ ] Other (Role, Template, Test, Documentation, etc.): **`{GoodieRole/Templates/Tests/Docs}: {Short Description}`**

###### Description of changes

A cheat sheet for the Catalan language.

###### Which issues (if any) does this fix?

###### People to notify (@mention interested parties)

---

Instant Answer Page: https://duck.co/ia/view/catalan_cheat_sheet

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @mention

daxtheduck commented:
@rubikkk Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

daxtheduck commented:
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

daxtheduck commented:
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Catalan Cheat Sheet)
                ![](https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_index.png)
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_mobile.png)

tagawa commented:
Hi @rubikkk, thanks very much for submitting this. As you may have seen, it's been installed to the beta server for testing: https://beta.duckduckgo.com/?q=catalan+cheat+sheet&ia=cheatsheet
We'll ask for other Catalan speakers to review it first before merging.

tagawa commented:
@rubikkk We have some feedback from Twitter - please could the upside-down exclamation marks and question marks be removed from the start of phrases? Apparently they should only appear at the end in Catalan, unlike Spanish: https://twitter.com/Xaviju/status/753570372906262528

Xaviju commented:
There is also a small typo:
```¿On estÀ el lavabo?`` should be ```On està el lavabo?```

rubikkk commented:
completely agree, I correct the typos and I also add a few more sentences
and I recommit the code.

thanks!

david.

On Thu, Jul 14, 2016 at 3:15 PM, Daniel Davis <notifications@github.com>
wrote:

> @rubikkk <https://github.com/rubikkk> We have some feedback from Twitter
> - please could the upside-down exclamation marks and question marks be
> removed from the start of phrases? Apparently they should only appear at
> the end in Catalan, unlike Spanish:
> https://twitter.com/Xaviju/status/753570372906262528
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3384#issuecomment-232661886>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ATeE-pkkVYCMF3jtX6ZbWzzODkxeVx9wks5qVjZzgaJpZM4JLLZ8>
> .
>

mcornella commented:
The `where is` forms are translated to "_on **és**_" (ON EHS) in Catalan. There is a little bit of confusion between when to use `és` (verb `ser`) vs. `està` (verb `estar`); in this case it's clear though: when talking about location you should use the verb `ser`. Some dialects of Catalan do use `estar` in all cases though, like Valencian.

So the following examples should be:
```
Where is the bathroom?
On és el lavabo?
ON EHS el la-BA-boo
```
```
Where is my room?
On és la meva habitació?
ON EHS la me-bah ah-bee-ta-SIOH
```

`EHS` is pronounced like the spelling of the letter `s`.

The usage of `ser` / `estar` is complicated though, and it's a big source of confusion even between native speakers. Hope that helps!

Xaviju commented:
Valencians we say ```estar``` but is recommended only for colloquial use. Both are correct.
Reference: chapter 33 Gramática normativa valenciana. Link to PDF: http://www.avl.gva.es/dms/GNV

mcornella commented:
Thanks! I read that it was more like a norm in Valencian, so the clarification helps.

rubikkk commented:
I corrected everything and added some bonus tracks ;)

commiting!

d.

On Thu, Jul 14, 2016 at 4:57 PM, Marc Cornellà <notifications@github.com>
wrote:

> Thanks! I read that it was more like a norm in Valencian, so the
> clarification helps.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3384#issuecomment-232690267>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ATeE-oErqrUBrLxkNN5aA29AwdqHWKJkks5qVk5DgaJpZM4JLLZ8>
> .
>

rubikkk commented:
i actually did it now! :)

rubikkk commented:
parsing errors, sorry no sleep :( corrected.

daxtheduck commented:
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Catalan Cheat Sheet)
                ![](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_index.png?AN8teDDleQ)
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_mobile.png?AN8teDDleQ)

rubikkk commented:
when I duckpan test I get this error:


README.md
catalan.json
Terminal
 ×
spanish.json
 ×


# Testing catalan.................. 1 FAILURE                                                                                                                                  
#        -> FAIL: invalid JSON                                                                                                                                                 
                                                                                                                                                                               
    #   Failed test 'FAIL: invalid JSON'                                                                                                                                       
    #   at t/CheatSheets/CheatSheetsJSON.t line 274.                                                                                                                           
    # Looks like you failed 1 test of 1.                                                                                                                                       
                                                                                                                                                                               
#   Failed test 'can_build'                                                                                                                                                    
#   at t/CheatSheets/CheatSheetsJSON.t line 275. 

and I don't know how to get it corrected.

but anyway, I uploaded a new version of the code with the fixes and extra fields, and looks like you didn't get to see it, i will commit again and see..

ManrajGrover commented on share/goodie/cheat_sheets/json/language/catalan.json:
```diff
@@ -0,0 +1,113 @@
+			"val": "On és la meva habitació?",
+			"trn": "ON ehs lah me-bah ah-bee-tah-SIOH"
+		}],
+		"Food and drink": [{
```
This should be `Food and Drink` (Notice the upper-cased D). It should match with what is mentioned in `section_order`.

ManrajGrover commented:
@rubikkk I've left a comment. Try fixing it. That should make the build pass. :smile:

rubikkk commented:
@ManrajGrover oh I didn't notice it! Thanks, fixing it in 3, 2, 1...! :)

daxtheduck commented:
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Catalan Cheat Sheet)
                ![](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_index.png?9WGHu24sHA)
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_mobile.png?9WGHu24sHA)

daxtheduck commented:
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Catalan Cheat Sheet)
                ![](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_index.png?nK7nH41WvR)
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_mobile.png?nK7nH41WvR)

daxtheduck commented:
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Catalan Cheat Sheet)
                ![](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_index.png?6HjHaJf61I)
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_mobile.png?6HjHaJf61I)

tagawa commented on share/goodie/cheat_sheets/json/language/catalan.json:
```diff
@@ -0,0 +1,113 @@
+				"trn": "GRA-see-əhs"
+			},
+
+			{
```
Please could you remove this empty line here?

tagawa commented on share/goodie/cheat_sheets/json/language/catalan.json:
```diff
@@ -0,0 +1,113 @@
+			}, {
+				"key": "Good Morning.",
+				"val": "Bon dia.",
+				"trn": "ah-DEhoo"
```
This should be "bon dee-ah"

tagawa commented on share/goodie/cheat_sheets/json/language/catalan.json:
```diff
@@ -0,0 +1,113 @@
+			}, {
+				"key": "Good Evening.",
+				"val": "Bona tarda.",
+				"trn": "ah-DEhoo"
```
This should be "bon-ah tar-dah"

mcornella replied:
Also, "evening" is `vespre`, you should use "Good Afternoon" instead.

tagawa commented on share/goodie/cheat_sheets/json/language/catalan.json:
```diff
@@ -0,0 +1,113 @@
+			}, {
+				"key": "Good Night.",
+				"val": "Bona nit.",
+				"trn": "ah-DEhoo"
```
This should be "bon-ah neet"

tagawa commented on share/goodie/cheat_sheets/json/language/catalan.json:
```diff
@@ -0,0 +1,113 @@
+			"val": "Vi",
+			"trn": "Bee"
+		}, {
+			"key": "Bread with tomatoe",
```
Small typo: "tomatoe" should be "tomato"

tagawa commented on share/goodie/cheat_sheets/json/language/catalan.json:
```diff
@@ -0,0 +1,113 @@
+		}, {
+			"key": "Can I enter/come in?",
+			"val": "Puc passar?",
+			"trn": "Pook pah-SAH"
```
Please change to:
`"val": "Que puc passar?",`
`"trn": "Kuh pook pah-SAH"`

Xaviju replied:
I think is correct as it is, without ```Que```. 
This is a regional dialect expression, not common everywhere and not necessary to understand the sentence.

Correct
 ```
"key": "Can I enter/come in?",
"val": "Puc passar?",
"trn": "Pook pah-SAH"
```

tagawa replied:
Thanks for the explanation. I'll remove my original comment.

rubikkk replied:
thanks everybody for the comments, all updated and commited!

d.

On Fri, Jul 15, 2016 at 10:18 AM, Daniel Davis <notifications@github.com>
wrote:

> In share/goodie/cheat_sheets/json/language/catalan.json
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3384#discussion_r70936794>
> :
>
> > +		}, {
> > +			"key": "Can you help me?",
> > +			"val": "Em pots ajudar?",
> > +			"trn": "əm pohts ah-joo-DAH"
> > +		}, {
> > +			"key": "What is your name?",
> > +			"val": "Com et dius?",
> > +			"trn": "Kom ət deeoos"
> > +		}, {
> > +			"key": "What time is it?",
> > +			"val": "Quina hora és?",
> > +			"trn": "Kee-noh-REHS"
> > +		}, {
> > +			"key": "Can I enter/come in?",
> > +			"val": "Puc passar?",
> > +			"trn": "Pook pah-SAH"
>
> Thanks for the explanation. I'll remove my original comment.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3384/files/2e2ef09f0d300aee479051a3d17fd147f66c7626#r70936794>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATeE-v43aV-XraWE9XNLZUyyN2j8RVAjks5qV0JdgaJpZM4JLLZ8>
> .
>

tagawa commented:
@rubikkk I've added a few more comments from people on Twitter. Also, we've had these two suggested phrases. Please could you add them?

You're welcome => De res.
Prounced: Dah RAYS

How much does this cost? -> Quant costa això?
Pronounced: Koo-and cost-ah asho

daxtheduck commented:
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Catalan Cheat Sheet)
                ![](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_index.png?jpwylwisne)
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_mobile.png?jpwylwisne)

daxtheduck commented:
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Catalan Cheat Sheet)
                ![](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_index.png?IzWfUHtLkN)
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_mobile.png?IzWfUHtLkN)

tagawa commented:
Thank you @rubikkk! It looks good on the beta server: https://beta.duckduckgo.com/?q=Catalan+Cheat+Sheet&ia=cheatsheet
so let's give it a couple more days for final feedback and then we should be good to merge.

mcornella commented on share/goodie/cheat_sheets/json/language/catalan.json:
```diff
@@ -0,0 +1,119 @@
+			"val": "Vi",
+			"trn": "Bee"
+		}, {
+			"key": "Bread with tomatofafa",
```
Typo on `tomato`,

mcornella commented on share/goodie/cheat_sheets/json/language/catalan.json:
```diff
@@ -0,0 +1,119 @@
+				"key": "See you later.",
+				"val": "Fins després.",
+				"trn": "FEENS-dehs-prehs"
+			}
```
This section (basics) is differently indented than the other sections.

rubikkk replied:
corrected typo and idention!

d.

On Sun, Jul 17, 2016 at 5:32 PM, Marc Cornellà <notifications@github.com>
wrote:

> In share/goodie/cheat_sheets/json/language/catalan.json
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3384#discussion_r71084291>
> :
>
> > +				"key": "Good Morning.",
> > +				"val": "Bon dia.",
> > +				"trn": "bon DEE-a"
> > +			}, {
> > +				"key": "Good Afternoon.",
> > +				"val": "Bona tarda.",
> > +				"trn": "bon-ah TAR-dah"
> > +			}, {
> > +				"key": "Good Night.",
> > +				"val": "Bona nit.",
> > +				"trn": "bon-ah NEET"
> > +			}, {
> > +				"key": "See you later.",
> > +				"val": "Fins després.",
> > +				"trn": "FEENS-dehs-prehs"
> > +			}
>
> This section (basics) is differently indented than the other sections.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/duckduckgo/zeroclickinfo-goodies/pull/3384/files/1b29346f96f66e656864d7ae1f6dfb866d98ada3#r71084291>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ATeE-kE7d5h_t-EgOdNrAzPB3fjnFmC8ks5qWksagaJpZM4JLLZ8>
> .
>

daxtheduck commented:
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Catalan Cheat Sheet)
                ![](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_index.png?Vv8OnNFveu)
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_mobile.png?Vv8OnNFveu)

daxtheduck commented:
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Catalan Cheat Sheet)
                ![](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_index.png?KUZ7iGbHL6)
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/catalan_cheat_sheet_mobile.png?KUZ7iGbHL6)

tagawa commented:
Thank you reviewers for the very helpful comments, and to @rubikkk for implementing them all. 
After all the feedback, it looks like everything's good on the beta server so let's merge this. It should go live in a few days.
Thanks again everyone! 🎉
